### PR TITLE
WIP: Add E2E tests

### PR DESCRIPTION
**Pull request by leonardola**
Can only be reviewed after #2238 gets merged

This is a Work in Progress PR so we can get feedback on the flows we are writing

## Changes proposed in this Pull Request:
This PR proposes to add E2E tests to the repository as there were none before. This way we can catch more errors before release and depend less on manual tests. Right now we have setup the environment and started writing some test flow.

Things to be discussed: 
- In WC Pay repo all the test flows are stored in a single file and inside merchant and shopper objects. That way the file becomes big really fast. So I decided to split the flows by area. Right now we have a `stripeUPESettingsUtils` object to store UPE related settings flows. Later we will have objects for non UPE settings, Other settings, Checkout, Pay Order page, My account, etc. This way we can separate flows into smaller files and objects.

- What flows should be added and what flows are the most important ones. Discussion at pdkmC9-7i-p2

## Testing instructions
No testing instructions for now. This is just a feedback request

### CODE REVIEW

**Comment by lovo-h** [comment]
> In WC Pay repo all the test flows are stored in a single file... That way the file becomes big really fast. So I decided to split the flows by area.

I agree that splitting the test flows into different files makes sense, in order to create more maintainable test files. As for splitting the test flow by area, I am not entirely sure about this approach but the basis for the decision makes sense. 

Although, one thing I want to avoid is having too many files because then, instead of a really big file, we will end up with a directory with a lot of small files. Which may also be detrimental. Thoughts?

> What flows should be added and what flows are the most important ones. Discussion at pdkmC9-7i-p2

I went ahead and summarized what flows we should prioritize here: pdkmC9-7i-p2#comment-196.

File: tests/e2e/specs/merchant/UPESettings.test.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { merchant } from '@woocommerce/e2e-utils';
```
Comment (cesarcosta99) [inline tests/e2e/specs/merchant/UPESettings.test.js]: I'd suggest renaming the file and use kebab case (`upe-settings.test.js`), so we can keep consistency across the JS file names.

File: tests/e2e/specs/merchant/example.test.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/**
```
Comment (cesarcosta99) [inline tests/e2e/specs/merchant/example.test.js]: Let's not forget to remove this file 😄

File: tests/e2e/utils/UPESettings.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
+import config from 'config';
+import { buttonsUtils } from './buttons';
+
+const baseUrl = config.get( 'url' );
+
+const UPE_SETTINGS_PAGE =
+	baseUrl + 'wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe';
```
Comment (cesarcosta99) [inline tests/e2e/utils/UPESettings.js]: Perhaps we can consider moving this into `tests/e2e/constants/screens.js` file or something like that, to gather all the screens we'll be navigating over the tests.

File: tests/e2e/utils/UPESettings.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
+import config from 'config';
```
Comment (cesarcosta99) [inline tests/e2e/utils/UPESettings.js]: Same about the kebab case file name.

File: tests/e2e/utils/UPESettings.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+const baseUrl = config.get( 'url' );
+
+const UPE_SETTINGS_PAGE =
+	baseUrl + 'wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe';
+
+export const stripeUPESettingsUtils = {
+	/**
+	 * Opens Upe settings page
+	 */
+	openSettingsPage: async () => {
```
Comment (cesarcosta99) [inline tests/e2e/utils/UPESettings.js]: I'd recommend explicitly export the functions instead exporting an object with function properties. That way we can import only the functions we need and declare and use them more easily.

File: tests/e2e/utils/buttons.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+export const buttonsUtils = {
+	clickButtonWithText: async ( text, parent = '' ) => {
```
Comment (cesarcosta99) [inline tests/e2e/utils/buttons.js]: Same about explicitly exporting the functions.

**Review by cesarcosta99 — COMMENTED** [review]
This is looking nice and it's very promising, can't wait to have it fully integrated into our test suite 😃 Thanks @a-danae and @leonardola for kicking this off!

I've left a few suggestions in a project structure level, they're mostly about being consistent while adding new code.

**Comment by lovo-h** [comment]
Adding the blocked label back in since the tests in this PR need to be updated to reflect the changes that the repo underwent since these tests were first written. 

@leonardola, thanks for letting me know that you will try to update these tests, hopefully, during the next cycle cooldown.

**Comment by diegocurbelo** [comment]
Closing this PR in favor of the recent Playwright E2E implementation.

**Pull request closed without merging**